### PR TITLE
add multi-arch nix support

### PR DESCRIPTION
**Pull request by DieracDelta**
I wanted to build this on aarch64-darwin. Had to bump nixpkgs (was getting a missing `lib` arg error) and refactor a bit.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -1,144 +1,148 @@
...
+    {
+
+      packages = forAllSystems (system: { inherit (nixpkgsFor.${system}) commonStyles storyBookYarnPkg storyBook; });
+
+      defaultPackage = forAllSystems (system: self.packages.${system}.commonStyles);
 
-    in flake;
+      checks = forAllSystems (system: { build = self.defaultPackage.${system}; });
+      #
+      # packages."${system}" = {};
```
Comment (gytis-ivaskevicius) [inline flake.nix]: Maybe this should be removed?

File: flake.nix [hunk 2]
```diff
@@ -1,144 +1,148 @@
...
+          buildInputs =
+            [
+              final.pkgs.storyBookYarnPkg
+            ] ++
+            (with prev.pkgs; [
+              nodejs
+            ]) ++
+            (with prev.pkgs.nodePackages; [
+              yarn
+            ]);
```
Comment (gytis-ivaskevicius) [inline flake.nix]:
```suggestion
            [
              final.pkgs.storyBookYarnPkg
              prev.nodejs
              prev.nodePackages.yarn
            ];
```

**Review by gytis-ivaskevicius — APPROVED** [review]

File: flake.nix [hunk 3]
```diff
@@ -1,144 +1,142 @@
 {
   description = "Common styles for NixOS related web sites.";
 
-  inputs.nixpkgs = { url = "nixpkgs/nixos-unstable"; };
+  inputs.nixpkgs = { url = "github:nixos/nixpkgs/nixos-21.11"; };
```
Comment (garbas) [inline flake.nix:4]: No need to switch away from unstable

File: flake.nix [hunk 4]
```diff
@@ -1,144 +1,142 @@
...
-      package = builtins.fromJSON (builtins.readFile ./package.json);
+      supportedSystems = [ "x86_64-linux" "aarch64-darwin" "x86_64-darwin" ];
 
-      flake = {
+      forAllSystems = f: nixpkgs.lib.genAttrs supportedSystems (system: f system);
+      nixpkgsFor = forAllSystems (system: import nixpkgs { inherit system; overlays = [ overlay ]; });
+      package = builtins.fromJSON (builtins.readFile ./package.json);
 
-        defaultPackage."${system}" = flake.packages."${system}".commonStyles;
+      overlay = final: prev: {
```
Comment (garbas) [inline flake.nix:17]: If we don't expose overlay in flake we don't need to use it and we can just deal directly with the package.

File: flake.nix [hunk 5]
```diff
@@ -1,144 +1,142 @@
...
+        commonStyles = prev.pkgs.stdenv.mkDerivation {
```
Comment (garbas) [inline flake.nix:18]: `prev` is pkgs, so `prev.pkgs....` makes little sense.

**Review by garbas — CHANGES_REQUESTED** [review]
@DieracDelta @gytis-ivaskevicius First thank you for filing a PR. I really appreciate it. This is also the reason why I left the comment in hope to give you some feedback on what could you do differently.

But I'm going to close this PR in favor of the [PR that improves also other things](https://github.com/NixOS/nixos-common-styles/pull/11) as well as adds support for multiple platforms.

I hope there are no bad feelings and if you need help how to contribute more please feel free to ping me on Matrix.

**Pull request closed without merging**